arbstate/inbox.go

```diff
@@ -127,10 +127,10 @@ func ParseSequencerMessage(ctx context.Context, batchNum uint64, batchBlockHash
 
 	// Stage 3: Decompress the brotli payload and fill the parsedMsg.segments list.
 	if len(payload) > 0 && daprovider.IsBrotliMessageHeaderByte(payload[0]) {
-		decompressed, err := arbcompress.Decompress(payload[1:], MaxDecompressedLen)
+		decompressed, err := arbcompress.Decompress(payload[1:], int(arbitrumChainParams.MaxUncompressedBatchSize)) // #nosec G115
 		if err == nil {
 			reader := bytes.NewReader(decompressed)
-			stream := rlp.NewStream(reader, uint64(MaxDecompressedLen))
+			stream := rlp.NewStream(reader, 0)
```

Comment:
I think this is strictly an improvement because now the stream will use the size of the underlying buffer (returned by decompress) and now will do better bounds checking when checking for space for segments.